Data #50

sdiby opened the pull request:
@ptatian Peter, I've created new vars in the ACS for cost burden, gross rent, and employed persons by major industries. I combined a few of the gross rent levels, but can parse those out if you'd prefer.

ptatian commented:
Thank you, @sdiby. I will review.

ptatian commented on Macros/ACS_summary_geo_source.sas:
```diff
@@ -2657,6 +2716,23 @@
+	NumRentSevereCostBurden_&_years. = B25070e10
+	NumOwnerCostBurden_&_years. = sum(B25091e8, B25091e9, B25091e10, B25091e11, B25091e19, B25091e20, B25091e21, B25091e22)
+	NumOwnSevereCostBurden_&_years. = sum(B25091e11, B25091e22)
+
```
We need to add vars for the denominators for these cost burden calculations. The denominator is all renter and all owner units, respectively, excluding the "Not computed" cells.

ptatian commented on Macros/ACS_summary_geo_source.sas:
```diff
@@ -1947,6 +1962,22 @@
 
+
+		mPopEmployedByInd_&_years. = C24030m1;
+			mPopEmployedAgric_&_years. = %moe_sum( var=C24030m3, C24030m30 );
```
There should not be any commas inside the var= parameter. Example:

`mPopEmployedAgric_&_years. = %moe_sum( var=C24030m3 C24030m30 );`

ptatian commented on Macros/ACS_summary_geo_source.sas:
```diff
@@ -2681,10 +2757,27 @@
     
     mNumRenterHsgUnits_&_years. = %moe_sum( var=mNumRenterOccupiedHU_&_years. mNumVacantHUForRent_&_years. );
 	
+	mNumRenterCostBurden_&_years. = %moe_sum( var=B25070m7, B25070m8, B25070m9, B25070m10)
```
Similar comment as above. No commas after var=.

ptatian commented on Macros/Acs_sf.sas:
```diff
@@ -93,6 +93,7 @@
   03/04/14 MSW Added sequence numbers 0063, 0064 and 0075. Added  B01002Ee/m, B25014Ee/m, and B19013Ee/m to drop list. 
   11/05/14 SXZ Added Table B19113 B09001
   06826/17 LP and RP Added Tables for racial equity work.
+  07/26/17 Somala Diby added B25106 for Bridge Park tabulations.
```
I guess you didn't actually have to modify this program.

sdiby commented:
Thanks @ptatian I've committed the above changes to this branch under "Responding to @ptatian 's changes," ready for your review.

ptatian commented:
@sdiby There are still a few syntax errors. 

```
MPRINT(ACS_SUMMARY_GEO_SOURCE):   mPopEmployedEduction_2011_15
NOTE 138-205: Line generated by the macro variable "_YEARS".
89             mPopEmployedEduction_2011_15
               ____________________________
               22
ERROR 22-322: Syntax error, expecting one of the following: !, !!, &, *, **, +, -, /, ;, <, <=, <>, =, >, ><, >=, AND, EQ, 
              GE, GT, IN, LE, LT, MAX, MIN, NE, NG, NL, NOTIN, OR, ^=, |, ||, ~=.  
```
You can test the program by submitting D:\DCData\Libraries\ACS\Prog\SF_2011_15\ACS_2011_15_dc_sum_all.sas. It's OK to submit the code as long as you are doing an **interactive SAS session** or a **batch submit from D:** because no files will be replaced on L:.

ptatian commented:
Sorry, @sdiby. I forgot to note earlier that it appears that the Bridge Park summary is not included in the list of summary files being created. In D:\DCData\Libraries\ACS\Macros\ACS_summary_geo_source.sas, you need to add the following at line 2889:

`    %ACS_summary_geo( bpk, &source_geo )`

sdiby commented:
Thanks @ptatian  I've added Bridge Park line to the list of summary files via commit.

ptatian commented:
Thanks, @sdiby, but I'm still getting errors. You have a number of statements with missing semicolons in Acs_summary_source_geo, for example: 

`	OwnerCostBurdenDenom_&_years. = NumOwnerOccupiedHU_&_years. - sum(B25091e12, B25091e23)`

Also, just realized another thing. Since we will be updating data sets that were created previously, we should include an appropriate note in the metadata. For example, in ACS_2011_15_dc_sum_all.sas, we should add a revisions= line to the %ACS_summary_all() macro invocation:

```
%ACS_summary_all( 

  /** State abbreviation. Ex: DC **/
  state_ab = DC,

  /** Year range (xxxx_yy). Ex: 2005_09 **/
  years = 2011_15,

  revisions=Add workers by industry/occupation, renters by gross rent, and housing cost burdens.

)
```

**Please note the comma (,) after years=2011_15.**

ptatian commented:
@sdiby Please add the revisions= line to **ACS_2011_15_va_sum_all.sas** and **ACS_2011_15_md_sum_all.sas** as well. Thanks!

ptatian commented:
Sigh...I should have written:

```
%ACS_summary_all( 

  /** State abbreviation. Ex: DC **/
  state_ab = DC,

  /** Year range (xxxx_yy). Ex: 2005_09 **/
  years = 2011_15,

  revisions=%str(Add workers by industry/occupation, renters by gross rent, and housing cost burdens.)

)
```

sdiby commented:
@ptatian @rpitingolo 
Rob, these changes are ready for your review:
- added missing semicolons in summary_geo_source macro
- added missing commas to sum_all progs
- added revisions line to sum_all_progs

rpitingolo commented:
@sdiby I'll review.

rpitingolo commented:
@sdiby I fixed some typos that were causing errors. I'm getting a " Variable B2506314 is uninitialized." note. It looks like you need these cells for some of your new variables. Was it added correctly in the SF programs?

sdiby commented:
@rpitingolo I've fixed the variable--it was read "B2506314" and should've read "B25063e14". I also fixed the corresponding MOE var name.

rpitingolo commented:
@sdiby OK. Glad it wasn't too complicated. Please go ahead and merge/finalize.
